providers/azure/resources/compute.go

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	compute "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/providers-sdk/v1/util/convert"
+	"go.mondoo.com/cnquery/providers/azure/connection"
+	"go.mondoo.com/cnquery/types"
+)
+
+func (a *mqlAzureSubscriptionCompute) id() (string, error) {
+	return "azure.subscription.compute/" + a.SubscriptionId.Data, nil
+}
+
+func initAzureSubscriptionCompute(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if len(args) > 0 {
+		return args, nil, nil
+	}
+
+	conn := runtime.Connection.(*connection.AzureConnection)
+	args["subscriptionId"] = llx.StringData(conn.SubId())
+
+	return args, nil, nil
+}
+
+func (a *mqlAzureSubscriptionCompute) vms() ([]interface{}, error) {
```

Comment:
i'm surprised to not see init functions for the vms, like initAzureSubscriptionComputeVm ?

Reply:
there's no discovery yet in this PR so that's why there are no init funcs for all of those assets, i will do this is a follow up with the discovery